### PR TITLE
fix: log to console using structured logging

### PR DESCRIPTION
**Pull request by minherz**
Replace package `console-log-level` with `@google-cloud/logging-min`. Use LogSync class of the logging package to print structured logging to `stdout` to capture the correct severity level.
Also handle a case of printing multi-line log that is captured as multiple log entries.

### CODE REVIEW

**Comment by minherz** [comment]
@bcoe can you help me to troubleshoot the reasons for check failures, please?

**Comment by minherz** [comment]
Test fails will be fixed with https://github.com/mapbox/node-pre-gyp/issues/623

File: src/logger.ts [hunk 1]
```diff
@@ -12,36 +12,69 @@
...
+  }
+
+  error(msg: string) {
+    if (this.severityThreshold > 0) {
+      this.log.error(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  private toOneLine(msg: string): string {
+    return msg.replace('\n', '\\n');
```
Comment (losalex) [inline src/logger.ts:74]: Do you need to handle '\r' as well?

File: src/logger.ts [hunk 2]
```diff
@@ -12,36 +12,69 @@
...
+    }
+  }
+
+  error(msg: string) {
+    if (this.severityThreshold > 0) {
+      this.log.error(this.log.entry(this.toOneLine(msg)));
+    }
+  }
+
+  private toOneLine(msg: string): string {
```
Comment (losalex) [inline src/logger.ts:73]: Is one line a requirement?

File: src/logger.ts [hunk 3]
```diff
@@ -12,36 +12,69 @@
...
-    level = 4;
+export class Logger {
+  private log: LogSync;
+  private severityThreshold: number;
+
+  constructor(readonly level?: number) {
+    if (level === undefined) {
+      level = defaultConfig.logLevel;
+    }
+    if (level < 0) {
```
Comment (losalex) [inline src/logger.ts:40]: I guess you should define a constants for this

Comment (losalex) [inline src/logger.ts:40]: Is this intentionally? I wonder in case of "invalid" value should you use `defaultConfig.logLevel`?

**Review by losalex — APPROVED** [review]
Some comments inside

File: src/logger.ts (hunk 1 above)
Comment (minherz) [reply src/logger.ts:74]: I do not think that standalone `\r` should be handled. Would the following handle Windows new lines IYO:
```java
var temp = msg.replace('\r\n', '\\r\\n');
return temp.replace('\n', '\\n');
```

File: src/logger.ts (hunk 2 above)
Comment (minherz) [reply src/logger.ts:73]: Yes. Printing to STDOUT is captured one line per log entry.

File: src/logger.ts (hunk 3 above)
Comment (minherz) [reply src/logger.ts:40]: Can you be more specific? What values should be set to constants?

Comment (minherz) [reply src/logger.ts:40]: I am not sure what you mean here. If the level is undefined then the default level is used. The conditions for level value beyond the range are handled exactly the same way they were handled before the change. The change does not modify existing interfaces or behavior, it only replaces the engine that prints logs to STDOUT behind it.

**Review by losalex — APPROVED** [review]
Left some comments, see if you agree with those

File: src/logger.ts (hunk 3 above)
Comment (minherz) [reply src/logger.ts:40]: It would change the current behavior.

File: src/logger.ts (hunk 1 above)
Comment (minherz) [reply src/logger.ts:74]: I saw it a common practice of capturing multiline payload in Json is to replace with an anchor that can be used to pretty print the text later.
It is hard to restore the original multi-line string later (e.g. callstacks). Such work will require from a reader a lot of effort and manual work. With the anchor, a simple find-and-replace in an editor or with a script/code will do the work.
Would removing \r means that in certain OS (e.g. Windows) restoring will not work?

Comment (minherz) [reply src/logger.ts:74]: Just updated the code. Please, see the new version that handles new lines in both Window and Linux formats.

File: src/logger.ts (hunk 3 above)
Comment (minherz) [reply src/logger.ts:40]: Added constants and a comment that explains where it comes from.

**Review by bcoe — APPROVED** [review]

**Pull request merged**